benchmark_controller.py

```diff
@@ -138,23 +141,27 @@ def _load_models(self):
 
         self._build_complete_flags()
 
-
-    def _post_run(self, stdout, perf_results):
+    def _post_run(self, benchmark_output):
         """This function executes after the benchmark has been run"""
-        self.logger.debug(stdout)
-        self.logger.debug(perf_results)
-        self._output_logs(stdout, perf_results)
+        for i in range(0, benchmark_output.len()):
+            stdout, perf_result, metadata = benchmark_output.get(i)
+            self._output_logs(stdout, perf_result, metadata)
         self.logger.info('The truth is out there')
 
     def _run_all(self, list_of_commands, perf=False):
         """This function executes the command required for the benchmark
         prebuild, build, postbuild, prerun and run. When perf needs to be ran,
         and output needs to be returned, set  perf to True"""
+        if perf:
```

Comment:
This is getting a bit silly. We shouldn't have conditional return and we now essentially have two independent paths in the same function.

Both perf and simple run should return a bundled output (out, err, meta), and it's up to the caller to know what to do with it.